### PR TITLE
[MM-0] - fix return wrapped error when net.Listen returns error

### PR DESCRIPTION
**Pull request by dmpichugin**
#### Summary

Fix wrap error
#### Ticket Link

N/A
#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by agnivade — APPROVED** [review]

File: app/server.go
Comment (isacikgoz) [inline app/server.go:1023]: Should we also change `api.server.start_server.starting.critical` to not use `%v` on translated text? Right now the code below prints duplicated error message.
```golang
err := errors.New("some error")
wErr := errors.Wrapf(err, utils.T("api.server.start_server.starting.critical"), err)
fmt.Println(wErr.Error())
```
output:
`Error starting server, err:some error: some error`

Alternatively, we can use `fmt.Errorf(utils.T("api.server.start_server.starting.critical"), err)` and change `%v` to `%w` in the translated text.

Comment (isacikgoz) [reply app/server.go:1021]: Created a [ticket](https://mattermost.atlassian.net/browse/MM-32274) for this 👍

**Review by isacikgoz — APPROVED** [review]
No need to block this one, LGTM 👍

**Pull request merged**
